File: src/main/java/com/intraedge/auth/service/ClientDetailsService.java
package com.intraedge.auth.service;

import java.util.List;

import com.intraedge.auth.dto.ClientDetails;


public interface ClientDetailsService {
	public List<ClientDetails> findAll();
	public ClientDetails findById(String id);
	public ClientDetails saveOrUpdate(ClientDetails dto);
  public void deleteById(String id);
  public void delete(ClientDetails dto);
  public void deleteAll(); 
}

File: src/main/resources/static/js/service/clientDetailsService.js
app.factory('clientDetailsService', ['$resource', 'appConfigService', function ($resource, appConfigService) {
	var baseUrl = appConfigService.protocol+"://"+appConfigService.host+":"+appConfigService.port+"";
	return $resource(baseUrl+'/auth-service/api/v1/clientdetails/:id', {}, {
        findAll: { method: 'GET', isArray: true },
        findById: { method: 'GET', params: {id: '@id'}},
        save: { method: 'POST',  headers: { 'Accept': 'text/plain' } },
        delete: { method: 'DELETE', params: {id: '@id'} }
    })
}]);